oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/name/ReadWriteNamespaceRegistry.java

```diff
@@ -38,8 +39,17 @@ public abstract class ReadWriteNamespaceRegistry
 
     private static final Logger LOG = LoggerFactory.getLogger(ReadWriteNamespaceRegistry.class);
 
-    public ReadWriteNamespaceRegistry(Root root) {
+    /**
+     * Feature flag to allow registering invalid namespace URIs (without a colon).
+     * Set the system property {@code oak.allowInvalidNamespaceUris} to {@code true} to enable this feature.
+     */
+    private final boolean allowInvalidNamespaceUris;
+
+    protected ReadWriteNamespaceRegistry(Root root) {
         super(root);
+        // cannot be static in order to allow testing with different values
```

Comment:
I don't understand how this is supposed to work. The system property value will not change for the given ClassLoader.

Reply:
The unit tests is reusing the same classloader to test with the feature being on and off. Therefore it cannot be static.